server/gateway.go

```diff
@@ -1563,6 +1566,8 @@ func (c *client) processGatewayRUnsub(arg []byte) error {
 		e = &outsie{ni: make(map[string]struct{}), sl: NewSublist()}
 		newe = true
 	}
+	e.Lock()
```

Comment:
ok will take a look.

Reply:
So should I do anything here? Or was that just an fyi?

Reply:
My comment was two folds:
- looks like you pick'ed up some changes (moving e.Lock()/e.Unlock()) that are currently not in master or in the PR I have. And know now why I moved it back: there will be a race if you leave it where you have moved it in line 1649 where we do `useSl = queue != nil || e.mode != modeOptimistic` because of the check on `e.mode`.
- Regardless of where we have e.Lock()/e.Unlock(), I am not sure it is good to be doing the defer. We have e.Lock() that is still be held when we release server lock to call `updateInterestForAccountOnGateway`.. so that's not good. My point is either to do the same e.Unlock()/e.Lock() around `updateInterestForAccountOnGateway` or manually unlocking everywhere in this function.

Reply:
Tell me what you prefer or feel free to change the code in gw to what you want and push an update. I am fine with either.

Reply:
Done. I have updated the PR.

Reply:
Thx